PoseEstimation-TFLiteSwift/OpenPose/OpenPosePoseEstimator.swift

```diff
@@ -193,29 +269,161 @@ private extension PoseEstimationOutput {
-            return Keypoint(position: CGPoint(x: x, y: y), score: score)
+            return Keypoint(column: keypointInfo.col,
+                            row: keypointInfo.row,
+                            width: OpenPosePoseEstimator.Output.ConfidenceMap.width,
+                            height: OpenPosePoseEstimator.Output.ConfidenceMap.height,
+                            value: keypointInfo.val)
         }
         
+        // Make [Human]
         return kps.map { keypoint in
-            let keypoints: [Keypoint] = OpenPosePoseEstimator.Output.BodyPart.allCases.enumerated().map { offset, _ in
-                return (offset == partIndex) ? keypoint : Keypoint(position: .zero, score: -100)
+            let keypoints: [Keypoint?] = OpenPosePoseEstimator.Output.BodyPart.allCases.enumerated().map { offset, _ in
+                return (offset == partIndex) ? keypoint : nil
             }
             return Human(keypoints: keypoints, lines: [])
         }
+    }
+    
+    func parseAllPartOnMultiHuman(from output: TFLiteFlatArray<Float32>, with threshold: Float?) -> [Human] {
+        let parts = OpenPosePoseEstimator.Output.BodyPart.allCases
+        var verticesForEachPart: [[(col: Int, row: Int, val: Float32)]?] = parts.map { _ in nil }
+        let pairs = OpenPosePoseEstimator.Output.BodyPart.lines
+        var edgesForEachPair: [[(from: (col: Int, row: Int, val: Float32), to: (col: Int, row: Int, val: Float32), cost: Float32)]] = pairs.map { _ in [] }
+        let (colSize, rowSize) = (OpenPosePoseEstimator.Output.ConfidenceMap.width,
+                                  OpenPosePoseEstimator.Output.ConfidenceMap.height)
         
-        // 2.
+        let pairCount = pairs.count
+        for (pairIndex, pair) in pairs.enumerated() {
+            // guard pairIndex == 0 else { continue }
+            
+            let startingPartIndex = pair.from.offsetValue()
+            let endingPartIndex = pair.to.offsetValue()
+            
+            let thresholdRatio: Float = 0.5 + (1.0 - (Float(pairIndex+1)/Float(pairCount))) * 0.5 // 1.0~>0.5
+            
+            // 1. Non Maximum Suppression, 2. Create Bipartite Graph
+            if verticesForEachPart[startingPartIndex] == nil {
+                verticesForEachPart[startingPartIndex] = output.keypoints(partIndex: startingPartIndex, filterSize: 5, threshold: threshold*thresholdRatio)
+            }
+            
+            // 1. Non Maximum Suppression, 2. Create Bipartite Graph
+            if verticesForEachPart[endingPartIndex] == nil {
+                verticesForEachPart[endingPartIndex] = output.keypoints(partIndex: endingPartIndex, filterSize: 5, threshold: threshold*thresholdRatio)
+            }
+            
+            guard let startingPartVertices = verticesForEachPart[startingPartIndex],
+                let endingPartVertices = verticesForEachPart[endingPartIndex] else { fatalError("cannot get verties for assignment") }
```

Comment:
can output.keypoints return nil?

Reply:
@syjdev 
No. `startingPartVertices` and `endingPartVertices` type's element can be `nil`. Because `startingPartVertices` and `endingPartVertices` are initialized with `nil` in fixed size.
But it's weird... I'll change `startingPartVertices` and `endingPartVertices` to non-nil element array and try to remove the `guard` statement.
👍